in-person-payments-example/Controllers/ApiController.cs

```diff
@@ -0,0 +1,273 @@
+            IPosAbortService posAbortService,
+            IPosTransactionStatusService posTransactionStatusService,
+            ITableService tableService, 
+            IOptions<AdyenOptions> options)
+        {
+            _logger = logger;
+            _posPaymentService = posPaymentService;
+            _posPaymentReversalService = posPaymentReversalService;
+            _posAbortService = posAbortService;
+            _posTransactionStatusService = posTransactionStatusService;
+            _tableService = tableService;
+            
+            _poiId = options.Value.ADYEN_POS_POI_ID;
+            _saleId = options.Value.ADYEN_POS_SALE_ID;
+        }
+
+        [HttpPost("api/create-payment")]
+        public async Task<ActionResult<CreatePaymentResponse>> CreatePayment([FromBody] CreatePaymentRequest request, CancellationToken cancellationToken = default)
+        {
+            var table = _tableService.Tables.FirstOrDefault(t => t.TableName == request.TableName);
+
+            if (table == null)
+            {
+                return NotFound(new CreatePaymentResponse()
+                {
+                    Result = "failure",
+                    RefusalReason = $"Table {request.TableName} not found"
+                });
+            }
+
+            string serviceId = IdUtility.GetRandomAlphanumericId(10);
+
+            table.PaymentStatusDetails.ServiceId = serviceId;
+            table.PaymentStatus = PaymentStatus.PaymentInProgress;
+
+            try
+            {
+                SaleToPOIResponse response = await _posPaymentService.SendPaymentRequestAsync(serviceId, _poiId,
+                    _saleId, request.Currency, request.Amount, cancellationToken);
+
+                PaymentResponse paymentResponse = response?.MessagePayload as PaymentResponse;
+                if (response == null)
+                {
+                    table.PaymentStatus = PaymentStatus.NotPaid;
+                    return BadRequest(new CreatePaymentResponse()
+                    {
+                        Result = "failure",
+                        RefusalReason = "Empty payment response"
+                    });
+                }
+
+                TableModel result = _tableService.UpdatePaymentStatus(table, paymentResponse);
+
+                if (result == null)
+                {
+                    table.PaymentStatus = PaymentStatus.NotPaid;
+                    return BadRequest(new CreatePaymentResponse()
+                    {
+                        Result = "failure",
+                        RefusalReason = _poiId == null
+                            ? "Could not reach payment terminal - POI ID is not set"
+                            : $"Could not reach payment terminal with POI ID {_poiId}"
+                    });
+                }
+
+                if (!string.IsNullOrWhiteSpace(result.PaymentStatusDetails.RefusalReason))
+                {
+                    return Ok(new CreatePaymentResponse()
```

Comment:
The response returns all those attributes but they are not used in the UI, maybe better to return only what it is used (we can log the extra fields in the ApiController)

Reply:
The transaction-status request now has this info on a separate page. I'm logging the response on the frontend

Reply:
I've removed the properties from the response. But now, I show them in the table itsself